test/Microsoft.AspNet.Mvc.Razor.Test/RazorPageTest.cs

```diff
@@ -687,96 +687,232 @@ public void IsSectionDefined_ThrowsIfPreviousSectionWritersIsNotRegistered()
-            var writer = new StringWriter();
+            var defaultTagHelperContent = new DefaultTagHelperContent();
+            var writer = new StringCollectionTextWriter(Encoding.UTF8);
             var context = CreateViewContext(writer);
-            var expectedContent = "Hello World!";
-            var contentToBeCopied = new DefaultTagHelperContent().SetContent("Hello ").Append("World!");
+            var tagHelperExecutionContext = new TagHelperExecutionContext(tagName: "p",
+                   selfClosing: false,
+                   items: new Dictionary<object, object>(),
+                   uniqueId: string.Empty,
+                   executeChildContentAsync: () => {
+                       defaultTagHelperContent.SetContent(input);
+                       return Task.FromResult(result: true);
+                   },
+                   startTagHelperWritingScope: () => { },
+                   endTagHelperWritingScope: () => defaultTagHelperContent);
+            tagHelperExecutionContext.Output =
+                new TagHelperOutput("p", new Dictionary<string, string>());
+            if (childContentRetrieved)
+            {
+                await tagHelperExecutionContext.GetChildContentAsync();
+            }
 
             // Act
             var page = CreatePage(p =>
             {
-                p.Write((ITextWriterCopyable)contentToBeCopied);
+                p.HtmlEncoder = new HtmlEncoder();
+                p.WriteTagHelperAsync(tagHelperExecutionContext).Wait();
             }, context);
             await page.ExecuteAsync();
 
             // Assert
-            Assert.Equal(expectedContent, writer.ToString());
+            Assert.Equal(expected, writer.ToString());
         }
 
         [Fact]
-        public async Task WriteTo_ITextWriterCopyable_WritesContent_ToSpecifiedWriter()
+        public async Task WriteTagHelperToAsync_WritesToSpecifiedWriter()
         {
             // Arrange
-            var writer = new StringWriter();
-            var expectedContent = "Hello World!";
-            var contentToBeCopied = new DefaultTagHelperContent().SetContent("Hello ").Append("World!");
+            var writer = new StringCollectionTextWriter(Encoding.UTF8);
+            var context = CreateViewContext(new StringWriter());
+            var tagHelperExecutionContext = new TagHelperExecutionContext(tagName: "p",
+                   selfClosing: false,
+                   items: new Dictionary<object, object>(),
+                   uniqueId: string.Empty,
+                   executeChildContentAsync: () => { return Task.FromResult(result: true); },
+                   startTagHelperWritingScope: () => { },
+                   endTagHelperWritingScope: () => new DefaultTagHelperContent());
+            tagHelperExecutionContext.Output =
+                new TagHelperOutput("p", new Dictionary<string, string>());
 
             // Act
             var page = CreatePage(p =>
```

Comment:
nit: lots of "p" in this test.  could use another lambda parameter such as `page` to differentiate it